Makefile

```diff
@@ -474,14 +476,29 @@ npm-update: node-check node_modules
 	npm install --package-lock
 
 .PHONY: js
-js: node-check $(JS_DEST)
+js: node-check fomantic $(JS_DEST)
 
 $(JS_DEST): node_modules $(JS_SOURCES)
 	npx eslint web_src/js webpack.config.js
 	npx webpack
 
+.PHONY: fomantic
+fomantic: $(FOMANTIC_DEST_DIR)
+node_modules/fomantic-ui/src: node-check node_modules
+
+# TODO(das7pad): replace with resolve.alias when building fomantic with webpack
+node_modules/fomantic-ui/src/theme.config: node_modules/fomantic-ui/src
+node_modules/fomantic-ui/src/theme.config: web_src/fomantic/theme.config.less
+ifeq (,$(wildcard $@))
+	cp $< $@
+endif
```

Comment:
Not sure I understand what's happening here or why this `ifeq` is not indented but wouldn't something like this be clearer?

```` make
node_modules/fomantic-ui/src/theme.config: node_modules web_src/fomantic/theme.config.less
  cp web_src/fomantic/theme.config.less $@

$(FOMANTIC_DEST_DIR): node_modules/fomantic-ui/src/theme.config
	npx gulp -f node_modules/fomantic-ui/gulpfile.js build

Reply:
`node_modules` is a phony target, hence it always runs and any following parents will always run too -- the fomantic assets are always rebuilt. We need an intermediate target that is not phony but backed by a file/dir on disk. 

There is a previous thread regarding the target body, but it is marked as outdated by Github: https://github.com/go-gitea/gitea/pull/9561#discussion_r363738468

Initially I symlinked the two - @lunny why was this changed in the first place?
>  why this `ifeq` is not indented

This will render the `...theme.config` target as empty using `Makefile` logic and skip the `cp` call.

Reply:
`cp` seems fine for platforms/file systems that do not support symlinks. It's more compatible that way.

Reply:
@silverwind this will copy the file only the source file changed(they are different).